engine/src/main/scala/cromwell/engine/workflow/tokens/TokenPool.scala

```diff
@@ -3,48 +3,19 @@ package cromwell.engine.workflow.tokens
 import java.util.UUID
 
 import cromwell.core.JobExecutionToken
-import JobExecutionToken.JobExecutionTokenType
-import cromwell.engine.workflow.tokens.TokenPool.TokenPoolPop
-
-import scala.language.postfixOps
-
-trait TokenPool {
-  def currentLoans: Set[JobExecutionToken]
-  def pop(): TokenPoolPop
-  def push(jobExecutionToken: JobExecutionToken): TokenPool
-}
+import cromwell.core.JobExecutionToken.JobExecutionTokenType
+import io.github.andrebeat.pool._
 
 object TokenPool {
-
-  case class TokenPoolPop(newTokenPool: TokenPool, poppedItem: Option[JobExecutionToken])
-
-  def apply(tokenType: JobExecutionTokenType): TokenPool = {
-    tokenType.maxPoolSize map { ps =>
-      val pool = (1 to ps toList) map { _ => JobExecutionToken(tokenType, UUID.randomUUID()) }
-      SizedTokenPool(pool, Set.empty)
-    } getOrElse {
-      InfiniteTokenPool(tokenType, Set.empty)
-    }
-  }
+  // 10 Million
+  val MaxCapacity = 10000000
```

Comment:
TOL The lack of separator support in 2.12 is annoying. I've been doing `10 * 1000 * 1000` for the time being until separator support supposedly lands in 2.13.